File: app/src/main/java/omniex/nl/omniex/ui/app/profile/order/ProfileOrderPresenter.java
package omniex.nl.omniex.ui.app.profile.order;

import javax.inject.Inject;

import nl.omniex.omniexshopping.data.repository.OrderRepository;
import nl.omniex.omniexshopping.ui.base.BasePresenter;

public class ProfileOrderPresenter extends BasePresenter<ProfileOrderView> {

    private OrderRepository mOrderRepository;

    @Inject
    ProfileOrderPresenter(OrderRepository orderRepository) {
        mOrderRepository = orderRepository;
    }

    void getOrderStatuses() {
        addDisposable(mOrderRepository
                .getOrderStatuses()
                .subscribe(
                        orderStatusesResponse -> {
                            if(orderStatusesResponse.code()==200)
                                ifViewAttached(view -> view.onOrderStatusesFetched(orderStatusesResponse.body().getOrderStatusList()));
                            }, Throwable::printStackTrace));
    }

}

File: app/src/main/java/omniex/nl/omniex/ui/app/profile/order/ProfileOrderPresenter.kt
package omniex.nl.omniex.ui.app.profile.order

import omniex.nl.omniex.data.repository.OrderRepository
import omniex.nl.omniex.ui.base.BasePresenter
import javax.inject.Inject

class ProfileOrderPresenter @Inject
internal constructor(private val mOrderRepository: OrderRepository) : BasePresenter<ProfileOrderView>() {

    fun getOrderStatuses() {
        addDisposable(mOrderRepository
                .orderStatuses()
                .subscribe(
                        { orderStatusesResponse ->
                            if (orderStatusesResponse.code() === 200)
                                ifViewAttached { view -> view.onOrderStatusesFetched(orderStatusesResponse.body()!!.orderStatusList!!) }
                        }, {it.printStackTrace()}))
    }

}
